news-bundle/src/Controller/NewsFeedController.php

```diff
@@ -0,0 +1,257 @@
+        'rss' => 'application/rss+xml',
+    ];
+
+    public function __construct(private ContaoFramework $framework, private FeedIo $feedIo, private ResponseTagger $responseTagger, private ContaoContext $contaoContext, private ImageFactory $imageFactory, private RequestStack $requestStack, private InsertTagParser $insertTags, private string $projectDir)
+    {
+    }
+
+    public function __invoke(Request $request, PageModel $pageModel): Response
+    {
+        $feed = new Feed();
+        $feed->setTitle($pageModel->title);
+        $feed->setDescription($pageModel->feedDescription);
+        $feed->setLanguage($pageModel->language);
+
+        $time = time();
+        $lastModified = null;
+
+        $archives = StringUtil::deserialize($pageModel->newsArchives);
+        $newsModel = $this->framework->getAdapter(NewsModel::class);
+
+        if ($pageModel->maxFeedItems > 0) {
+            $articles = $newsModel->findPublishedByPids($archives, null, $pageModel->maxFeedItems);
+        } else {
+            $articles = $newsModel->findPublishedByPids($archives);
+        }
+
+        // TODO: Is this still necessary?
+        $origObjPage = $GLOBALS['objPage'] ?? null;
+
+        foreach ($articles as $article) {
+            // Never add unpublished elements to the RSS feeds
+            if (!$article->published || ($article->start && $article->start > $time) || ($article->stop && $article->stop <= $time)) {
+                continue;
+            }
+
+            // TODO: Throw an event here to allow for custom model-to-item transformation?
+            $item = $this->transformModelToItem($article, $pageModel->feedSource, $pageModel);
+
+            if (null === $lastModified) {
+                $lastModified = $item->getLastModified();
+            }
+
+            if ($lastModified < $item->getLastModified()) {
+                $lastModified = $item->getLastModified();
+            }
+
+            $feed->add($item);
+        }
+
+        // TODO: Is this still necessary?
+        $GLOBALS['objPage'] = $origObjPage;
+
+        $feed->setLastModified($lastModified);
+
+        $response = new Response($this->feedIo->format($feed, $pageModel->feedFormat));
+        $response->setLastModified($feed->getLastModified());
+        $response->headers->set('Content-Type', $this->contentTypes[$pageModel->feedFormat]);
+
+        foreach ($archives as $archive) {
+            $this->responseTagger->addTags(['contao.db.news_archive.'.$archive]);
+        }
+
+        return $response;
+    }
+
+    public function configurePageRoute(PageRoute $route): void
+    {
+        $format = $route->getPageModel()->feedFormat;
+        $route->setUrlSuffix($this->urlSuffixes[$format]);
+    }
+
+    public function getUrlSuffixes(): array
+    {
+        return array_unique(array_values($this->urlSuffixes));
+    }
+
+    private function transformModelToItem(NewsModel $model, string $source, PageModel $pageModel): Item
```

Comment:
maybe call it `$feedSource` instead of `$source` to be consistent with page model and other methods